### PR TITLE
Port of https://github.com/johnlindquist/angularjs-plugin to yasnippets

### PR DESCRIPTION
**Pull request by lispnik**
Hi, Thanks for the great work angular-snippets.el. I've been using it and added in some JavaScript snippets. They are based on the ones from John Lindquist's IntelliJ plugin: https://github.com/johnlindquist/angularjs-plugin

### CODE REVIEW

File: snippets/js-mode/ngmfa.yasnippet [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: ngmfa
+# key: ngmfa
```
Comment (magnars) [inline snippets/js-mode/ngmfa.yasnippet:3]: Why not just `ngfa` ?

File: snippets/js-mode/ngmfi.yasnippet [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# -*- mode: snippet -*-
+# name: ngmfi
+# key: ngmfi
```
Comment (magnars) [inline snippets/js-mode/ngmfi.yasnippet:3]: `ngfi` ?

File: snippets/js-mode/ngdl.yasnippet [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# -*- mode: snippet -*-
+# name: ngdl
+# key: ngdl
```
Comment (magnars) [inline snippets/js-mode/ngdl.yasnippet:3]: What's the `l` for? Wound make more sense with either `ngdi` or just `ngd`?

**Comment by magnars** [comment]
Thanks for the contribution. :-)

I would like a more predictable pattern for the mnemonic shortcuts. Like `ngdl` is directive, but `ngmfa` is factory, while `ngms` is service. Maybe always `ng` + two first letters for these?

**Comment by lispnik** [comment]
I pretty much followed John Lindquist's abbreviations from his IntelliJ plugin since my fingers were already used to them (see https://github.com/johnlindquist/angularjs-plugin/blob/master/resources/liveTemplates/AngularJS.xml) A more consistent scheme makes sense though.

**Comment by magnars** [comment]
> A more consistent scheme makes sense though.

Is that something you'll fix for this PR, or should I?

**Pull request merged**